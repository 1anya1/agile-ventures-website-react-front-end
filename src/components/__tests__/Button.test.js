import React from 'react';
import { shallow } from 'enzyme';
import { Button } from 'react-bootstrap';

describe('Button', () => {
  it('should be defined', () => {
   expect(Button).toBeDefined();
  });

  it('should render correctly', () => {
   const tree = shallow(
     <Button>button test</Button>
   );
   expect(tree).toMatchSnapshot();
  });

  // it('renders without crashing', () => {
  //   shallow(<Button />);
  // });

  it('should have a button value', () => {
   const tree = shallow(
     <Button>button test</Button>
   );

   expect(tree.find('button')).toHaveLength(1);
   expect(tree.find('button').html()).toContain('button test');
 });
});
