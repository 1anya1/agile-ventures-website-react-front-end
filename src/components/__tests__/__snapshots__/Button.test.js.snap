// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    active={false}
    block={false}
    bsClass="btn"
    bsStyle="default"
    disabled={false}
>
    button test
</Button>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "button test",
      "className": "btn btn-default",
      "disabled": false,
      "type": "button",
    },
    "ref": null,
    "rendered": "button test",
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "button test",
        "className": "btn btn-default",
        "disabled": false,
        "type": "button",
      },
      "ref": null,
      "rendered": "button test",
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
